pkg/worker/events.go

```diff
@@ -0,0 +1,97 @@
+package worker
+
+import (
+	"log"
+	"time"
+
+	"github.com/beam-cloud/beta9/pkg/repository"
+	types "github.com/beam-cloud/beta9/pkg/types"
+	"github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func collectAndSendContainerMetrics(eventRepo repository.EventRepository, workerID string, request *types.ContainerRequest, spec *specs.Spec, pidChan <-chan int, done chan bool) {
+	gpuEnabled := request.Gpu != ""
+
+	ticker := time.NewTicker(types.ContainerResourceUsageEmissionInterval)
+	defer ticker.Stop()
+
+	containerPid := <-pidChan
+	if containerPid == 0 {
+		return
+	}
+
+	containerProc, err := NewProcUtil(containerPid)
+	if err != nil {
+		log.Printf("Unable to get process for container %s: %v\n", request.ContainerId, err)
+		return
+	}
+
+	var prevProcCpuTime float64 = 0
+	var prevSystemCpuTime float64 = 0
+	for {
+		select {
+		case <-done:
+			return
+
+		case <-ticker.C:
+			// Calculate GPU memory
+			var gpuMemUsed int64
+			var gpuMemTotal int64
+			if gpuEnabled {
+				for _, device := range spec.Linux.Resources.Devices {
+					stats, err := GetGpuMemoryUsage(*device.Minor)
```

Comment:
@luke-lombardi does this look right to you? I found that Minor should point to the device on disk as mentioned [here](https://man.archlinux.org/man/nvidia-smi.1.en#Minor_Number).

Reply:
Maybe we can get the device and store it when worker starts up. I can investigate when i start on the multi-worker gpu nodes

Reply:
This will go through all of the devices and attempt to use nvidia-smi to get this information. An error will be returned for those that don't work with nvidia-smi. This is kind of a brute force way of doing it, but I don't think we set `spec.Linux.Resources.Devices` for anything else at the moment. So this _should_ work.